### ISSUE TITLE
When CodeReady Workspaces is installed on the cluster - the link to load devfile should load the devfile from the appropriate branch

### ISSUE THREAD

**Issue by edeandrea** [openshift/console#6077]
Cross-posted also as https://github.com/eclipse/che/issues/17448

### Steps to reproduce
1. Log into OpenShift cluster as cluster admin
1. Deploy the Eclipse Che operator as well as a `CheCluster` instance, then wait for it to spin up
1. If logged in as administrator, on left side menu change 'adminstrator' to 'developer' to show `topology` link on left nav bar.
1. Click **+Add**
1. Click **From Git**
1. In the **Git Repo URL** field, enter https://github.com/edeandrea/summit-lab-spring-music
1. Expand **Show Advanced Git Options**
1. In the **Git Reference field**, enter **pipeline**
1. In the **Resources** section, select **Deployment Config**
1. In the **Advanced Options** section, click the word **Labels** and enter the label **app.openshift.io/runtime=spring** and hit enter
1. Click **Create**
1. Wait for build & deployment to complete
1. Click the **che** logo on the deployment (see attached screenshot)

### Expected behavior
After going through the steps to reproduce above, the URL that is generated for the icon is `https://<che-url>/dashboard/#/load-factory/?url=https://github.com/edeandrea/summit-lab-spring-music&policies.create=peruser` but it should be `https://<che-url>/dashboard/#/load-factory/?url=https://github.com/edeandrea/summit-lab-spring-music/tree/pipeline&policies.create=peruser` since the application was added using the **pipeline** branch. It should respect that and hence load the workspace using the devfile on the **pipeline** branch and not the **master** branch.

The correct annotations are set on the `DeploymentConfig`

```
app.openshift.io/vcs-ref: pipeline
app.openshift.io/vcs-uri: 'https://github.com/edeandrea/summit-lab-spring-music'
```

![87980389-8d74aa00-caa1-11ea-81c7-4a155a432ac5](https://user-images.githubusercontent.com/363447/88212390-31d32980-cc25-11ea-99b5-8302b4fbdecc.png)

**Comment by spadgett** [comment]
cc @christianvogt

**Comment by edeandrea** [comment]
Ping so the issue doesn't go stale

**Comment by rottencandy** [comment]
Fixed in #6764
